src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Registration/RegistrationIndexValidator.cs

```diff
@@ -18,34 +18,54 @@ public abstract class RegistrationIndexValidator : RegistrationValidator
         {
         }
 
-        protected override async Task<bool> ShouldRun(ValidationContext data)
+        protected override async Task<bool> ShouldRunAsync(ValidationContext context)
         {
-            return 
-                await base.ShouldRun(data) &&
+            var shouldRunTask = context.GetCachedResultAsync(
+                Keys.ShouldRunAsync,
```

Comment:
I'm not a big fan of this syntax:
- it requires the validators to know the cache exists
- it requires the validators to specify a lambda for the operation they want to do
- it requires the validators to specify the exact and correct key for the operation they want to do

Ideally, the validators shouldn't be aware of the context's cache. The context should everything automatically.

What if:
- the context had `Lazy`s for `V2RegistrationIndex`, `V2RegistrationLeaf`, `V3RegistrationIndex`, `V3RegistrationLeaf`, etc (anything that requires a web request)
- validators were rewritten to only access these fields instead of making web requests themselves

Another option:
- move methods to access `V2RegistrationIndex`, `V2RegistrationLeaf`, `V3RegistrationIndex`, `V3RegistrationLeaf`, etc to the context
- cache the returns in the context